crates/oxc_ecmascript/src/constant_evaluation/mod.rs

```diff
@@ -327,6 +327,10 @@ pub trait ConstantEvaluation<'a>: MayHaveSideEffects {
                     if matches!(name, "Object" | "Number" | "Boolean" | "String")
                         && self.is_global_reference(right_ident)
                     {
+                        let left_ty = ValueType::from(left);
+                        if left_ty.is_undetermined() {
+                            return None;
+                        }
```

Comment:
Without this change, this test failed because `var x; foo(!1)` was output.
https://github.com/oxc-project/oxc/blob/d6d13dd296fd365d4229e02d72a6658266b914c7/crates/oxc_minifier/src/peephole/fold_constants.rs#L1724